### ISSUE TITLE
MDC Tab Bar: Make manual activation the default behavior

### ISSUE THREAD

**Issue by kfranqueiro** [material-components/material-components-web#3286]
(See https://www.w3.org/TR/wai-aria-practices/#tabpanel for info and examples on automatic vs. manual activation.)

Currently the new MDC Tab Bar implements automatic activation (each tab activates as soon as you focus it with the arrow keys). However, this risks many "drive-by" requests in cases where the Tab Bar governs content that is asynchronously requested on Tab activation. The default behavior should be manual activation (press space/enter to activate a tab after focusing it with the arrow keys) to avoid risking this situation. Mouse/touch interaction behavior should remain unaffected.

If possible, it'd be nice to still provide an option for automatic activation.

**Comment by kfranqueiro** [comment]
The current implementation is going to need to be revised quite a bit to make this work...

Currently, both `mdc-tab-bar` and `mdc-tab` make assumptions that activation is directly tied to focus - so much so, that _activation_ actually drives the setting of the `tabindex` attribute (i.e. focus happens in reaction to activation, which is the reverse of how this should function). We're going to need to revise logic in at least 2 packages to accomplish this.
